sae/vm_test.go

```diff
@@ -270,3 +274,94 @@ func TestSyntacticBlockChecks(t *testing.T) {
 		})
 	}
 }
+
+func TestSemanticBlockChecks(t *testing.T) {
+	const now = 1e6
+	ctx, sut := newSUT(t, 1, blockstest.WithTimestamp(now))
+	sut.rawVM.config.Now = func() time.Time {
+		return time.Unix(now, 0)
+	}
+
+	lastAcceptedID, err := sut.LastAccepted(ctx)
+	require.NoError(t, err, "LastAccepted()")
+	lastAccepted, err := sut.GetBlock(ctx, lastAcceptedID)
+	require.NoError(t, err, "GetBlock(lastAcceptedID)")
+
+	tests := []struct {
+		name           string
+		parentHash     common.Hash
+		acceptedHeight bool
+		time           uint64
+		receipts       types.Receipts
+		wantErr        error
+	}{
+		{
+			name:       "unknown_parent",
+			parentHash: common.Hash{1},
+			wantErr:    errUnknownParent,
+		},
+		{
+			name:           "already_finalized",
+			acceptedHeight: true,
+			wantErr:        errFinalizedBlock,
+		},
+		{
+			name:    "block_time_under_minimum",
+			time:    1,
```

Comment:
```suggestion
			time:    saeparams.TauSeconds-1,
```

(readability) It explains why to the reader.